rust/hyperlane-base/src/db/rocks/hyperlane_db.rs

```diff
@@ -152,6 +154,22 @@ impl HyperlaneRocksDB {
         Ok(true)
     }
 
+    /// Store the merkle tree insertion event, and also store a mapping from message_id to leaf_index
+    pub fn process_tree_insertion(&self, insertion: &MerkleTreeInsertion) -> DbResult<bool> {
+        if let Ok(Some(_)) = self.retrieve_merkle_tree_insertion_by_leaf_index(&insertion.index()) {
```

Comment:
Iiuc this short circuiting means that double insertion won't work correctly? E.g. if index 10 and 11 are a double insertion, then index 10 will get stored, but then you will skip storing index 11, which means that the merkle tree processor logic here will break https://github.com/hyperlane-xyz/hyperlane-monorepo/pull/2746/files#diff-88d3caf1bed3eeed9ef0ee7e9ac5743c5e5c401e47c9ba77c6257e5ccb5f4a01R73

Reply:
the key here is the merkle tree leaf index rather than the message nonce, which doesn't skip on index 11. Or am I misunderstanding?

Reply:
Oh oops my bad, you're right